File: onyx-database/src/main/java/com/onyx/exception/RelationshipNotFoundException.java
package com.onyx.exception;

/**
 * Created by timothy.osborn on 1/1/15.
 *
 */
public class RelationshipNotFoundException extends EntityException
{
    public static final String RELATIONSHIP_NOT_FOUND = "Relationship not found: ";

    /**
     * Constructor with message
     *
     * @param message Error message
     */
    public RelationshipNotFoundException(String message, String relationship, String className)
    {
        super(message + relationship + " for class " + className);
    }

    @SuppressWarnings("unused")
    public RelationshipNotFoundException()
    {
        super();
    }
}

File: onyx-database/src/main/kotlin/com/onyx/exception/RelationshipNotFoundException.kt
package com.onyx.exception

/**
 * Created by timothy.osborn on 1/1/15.
 *
 */
class RelationshipNotFoundException(message: String, relationship: String, className: String) : OnyxException(message + relationship + " for class " + className) {

    companion object {
        @JvmField val RELATIONSHIP_NOT_FOUND = "Relationship not found: "
    }
}
